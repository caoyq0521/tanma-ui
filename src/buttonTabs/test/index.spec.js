import { mount } from '@vue/test-utils';
import TmButtonTabs from '../button-tabs.vue';

describe('tm-button-tabs', () => {
  it('render buttonTabs', () => {
    const wrapper = mount(TmButtonTabs);
    expect(wrapper).toMatchSnapshot();
  });

  it('create button-tabs', () => {
    const wrapper = mount(TmButtonTabs);

    expect(wrapper.find('.tm-button-tabs').exists()).toBe(true);
  });

  it('whether type of button', async () => {
    const wrapper = mount(TmButtonTabs);

    await wrapper.setProps({ type: 'step' });
    expect(wrapper.find('.tm-button-tabs__step').exists()).toBe(true);

    await wrapper.setProps({ type: 'certainty' });
    expect(wrapper.find('.tm-button-tabs__certainty').exists()).toBe(true);

  });

})

  