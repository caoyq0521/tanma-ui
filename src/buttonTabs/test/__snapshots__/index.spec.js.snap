// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tm-button-tabs render buttonTabs 1`] = `
<div class="tm-button-tabs tm-button-tabs__step"><button type="default" class="tm-button tm-button--medium tm-button--default tm-button-tabs__step-prve">
    <!----> 上一步
  </button> <button type="primary" class="tm-button tm-button--medium tm-button--primary tm-button-tabs__step-next">
    <!----> 下一步
  </button></div>
`;
